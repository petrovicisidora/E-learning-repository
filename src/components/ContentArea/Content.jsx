//Content component that contains image and its children

import React from "react";

import {
  StyledChildrenContainer,
  StyledContentContainer,
  StyledImageContainer,
} from "./Content.style";

function Content({ image, children }) {
  return (
    <StyledContentContainer>
      <StyledImageContainer src={image} alt="Content Image" />
      <StyledChildrenContainer>{children}</StyledChildrenContainer>
    </StyledContentContainer>
  );
}

export default Content;
