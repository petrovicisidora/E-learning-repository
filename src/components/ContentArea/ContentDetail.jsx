//Content Detail component with detailed information about each course

import React from "react";

import { renderCircles } from "./helper";
import {
  StyledContentDetailContainer,
  StyledContentProgress,
  StyledContentText,
  StyledContentTimeContainer,
  StyledContentTitle,
} from "./Content.style";

function ContentDetail({ title, time, text, progress }) {
  return (
    <StyledContentDetailContainer>
      <StyledContentTitle>{title}</StyledContentTitle>
      <StyledContentTimeContainer>
        <div>{time}</div>
        <div>To be done by:</div>
      </StyledContentTimeContainer>
      <div>
        <StyledContentProgress>{renderCircles(progress)}</StyledContentProgress>
        <StyledContentText>{text}</StyledContentText>
      </div>
    </StyledContentDetailContainer>
  );
}
export default ContentDetail;
