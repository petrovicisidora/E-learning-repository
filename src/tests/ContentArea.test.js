import React from "react";
import { render, screen } from "@testing-library/react";

import Content from "../components/ContentArea/Content";
import ContentDetail from "../components/ContentArea/ContentDetail";

const contentAreaProps = {
  image: "/path/to/image.png",
  children: (
    <ContentDetail
      title="Test Title"
      time="Test Time"
      text="Test Text"
      progress={2}
    />
  ),
};

test("renders content's image correctly", () => {
  render(<Content {...contentAreaProps} />);
  const contentAreaImage = screen.getByAltText("Content Image");

  expect(contentAreaImage).toBeInTheDocument();
});

test("renders content's children (ContentDetail) correctly", () => {
  render(<Content {...contentAreaProps} />);
  const contentDetailTitle = screen.getByText("Test Title");
  const contentDetailTime = screen.getByText("Test Time");

  expect(contentDetailTitle).toBeInTheDocument();
  expect(contentDetailTime).toBeInTheDocument();
});

test("content has correct styles", () => {
  render(<Content {...contentAreaProps} />);
  const contentImage = screen.getByAltText("Content Image");

  expect(contentImage).toHaveStyle("float: left");
  expect(contentImage).toHaveStyle("margin-right: 15px");
});

test("content detail (children) has correct styles", () => {
  render(
    <ContentDetail
      title="Test Title"
      time="Test Time"
      text="Test Text"
      progress={2}
    />
  );
  const contentDetailTitle = screen.getByText("Test Title");

  expect(contentDetailTitle).toHaveStyle("color: #303330a6");
  expect(contentDetailTitle).toHaveStyle("font-size: 20px");
});

test("matches snapshot", () => {
  const { container } = render(<Content />);
  expect(container).toMatchSnapshot();
});
