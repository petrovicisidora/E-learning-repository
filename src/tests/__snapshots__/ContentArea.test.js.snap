// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="sc-beySPh cCHNsi"
  >
    <img
      alt="Content Image"
      class="sc-guDLey hOyfhk"
    />
    <div
      class="sc-dmyCSP gCwXEZ"
    />
  </div>
</div>
`;
